students/ltrisolini/Lesson09/mailroom_OO.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+import sys
+from collections import defaultdict
+
+class Donor:
+
+    def __init__(self, name, donations=None):
+        self.name = name
+        if donations is None:
+            self.donations = []
+        else:
+            self.donations = donations
+
+
+    def add_donation(self, amount):
+        try:
+            self.donations.append(int(amount))
+        except ValueError:
+            print("Please enter a number!")
+
+
+    def number_donations(self):
+        return len(self.donations)
+
+
+    def total_donation(self):
+        try:
+            return sum(self.donations)
+        except TypeError:
+            return self.donations
+
+    def avg_donation(self):
+        try:
+            return self.total_donation()/ self.number_donations()
+        except TypeError:
+            return self.donations
+
+class Roster:
+
+    def __init__(self, donors=None):
+        self.donor_list = donors
+
+    def thank_you(self):
+        '''Accepts user input, and then adds donors / donations to the donor dictionary.
+        It also prints a thank you message for the latest donation'''
+        print('Please enter the donor name\n (Type "list" for a list of current donor names)\n '
+            'Press "q" to return to console')
+        new_donor = input(':')
```

Comment:
please review instructions - there should be no input functions; these are data classes and should not interact with the user.

separating "UI" code makes it easier to adjust those parts without changing business logic.